mmcls/models/backbones/levit.py

```diff
@@ -0,0 +1,545 @@
+    def train(self, mode=True):
+        super(AttentionSubsample, self).train(mode)
+        if mode and hasattr(self, 'ab'):
+            del self.ab
+        else:
+            self.ab = self.attention_biases[:, self.attention_bias_idxs]
+
+    def forward(self, x):
+        B, N, C = x.shape
+        k, v = self.kv(x).view(B, N, self.num_heads,
+                               -1).split([self.key_dim, self.d], dim=3)
+        k = k.permute(0, 2, 1, 3)  # BHNC
+        v = v.permute(0, 2, 1, 3)  # BHNC
+        q = self.q(x).view(B, self.resolution_2, self.num_heads,
+                           self.key_dim).permute(0, 2, 1, 3)
+
+        attn = (q @ k.transpose(-2, -1)) * self.scale + \
+               (self.attention_biases[:, self.attention_bias_idxs]
+                if self.training else self.ab)
+        attn = attn.softmax(dim=-1)
+
+        x = (attn @ v).transpose(1, 2).reshape(B, -1, self.dh)
+        x = self.proj(x)
+        return x
+
+
+@MODELS.register_module()
+class LeViT(BaseBackbone):
+    """Vision Transformer with support for patch or hybrid CNN input stage."""
+    arch_zoo = {
+        **dict.fromkeys(
+            ['128s'], {
+                'embed_dim': [128, 256, 384],
+                'num_heads': [4, 6, 8],
+                'depth': [2, 3, 4],
+                'key_dim': [16, 16, 16],
+                'down_ops': [['Subsample', 16, 128 // 16, 4, 2, 2],
+                             ['Subsample', 16, 256 // 16, 4, 2, 2]]
+            }),
+        **dict.fromkeys(
+            ['128'], {
+                'embed_dim': [128, 256, 384],
+                'num_heads': [4, 8, 12],
+                'depth': [4, 4, 4],
+                'key_dim': [16, 16, 16],
+                'down_ops': [['Subsample', 16, 128 // 16, 4, 2, 2],
+                             ['Subsample', 16, 256 // 16, 4, 2, 2]]
+            }),
+        **dict.fromkeys(
+            ['192'], {
+                'embed_dim': [192, 288, 384],
+                'num_heads': [3, 5, 6],
+                'depth': [4, 4, 4],
+                'key_dim': [32, 32, 32],
+                'down_ops': [['Subsample', 32, 192 // 32, 4, 2, 2],
+                             ['Subsample', 32, 288 // 32, 4, 2, 2]]
+            }),
+        **dict.fromkeys(
+            ['256'], {
+                'embed_dim': [256, 384, 512],
+                'num_heads': [4, 6, 8],
+                'depth': [4, 4, 4],
+                'key_dim': [32, 32, 32],
+                'down_ops': [['Subsample', 32, 256 // 32, 4, 2, 2],
+                             ['Subsample', 32, 384 // 32, 4, 2, 2]]
+            }),
+        **dict.fromkeys(
+            ['384'], {
+                'embed_dim': [384, 512, 768],
+                'num_heads': [6, 9, 12],
+                'depth': [4, 4, 4],
+                'key_dim': [32, 32, 32],
+                'down_ops': [
+                    ['Subsample', 32, 384 // 32, 4, 2, 2],
+                    ['Subsample', 32, 512 // 32, 4, 2, 2],
+                ]
```

Comment:
Since the downsample settings are the same among all architectures, better to remove them from the architectures.
As for the number of heads, since it's directly calculated from the embed dims, please calculate it when building the downsampling module.